infrastructure/instances.tf

```diff
@@ -243,98 +243,265 @@ data "template_file" "nomad_client_script_smasher_smusher" {
-
-    tag {
-        key = "Name"
-        value = "Nomad Client Instance ${var.user}-${var.stage}"
-        propagate_at_launch = true
+    tags {
+        Name = "Spot Fleet Launch Specification x1.32xlarge ${var.user}-${var.stage}"
+        User = "${var.user}"
+        Stage = "${var.stage}"
     }
+  
+  }
+
+  ##
+  # x1e.16xlarge
+  ##
+  launch_specification {
+
+    # Client Specific
+    instance_type             = "x1.16xlarge"
+    weighted_capacity         = 20 # via https://aws.amazon.com/ec2/instance-types/
+    spot_price                = "${var.spot_price}"
+    ami                       = "${data.aws_ami.ubuntu.id}"
+    iam_instance_profile_arn  = "${aws_iam_instance_profile.data_refinery_instance_profile.arn}"
+    user_data                 = "${data.template_file.nomad_client_script_smusher.rendered}"
+    vpc_security_group_ids    = ["${aws_security_group.data_refinery_worker.id}"]
+    subnet_id                 = "${aws_subnet.data_refinery_1a.id}"
+    availability_zone         = "${var.region}a"
+    key_name = "${aws_key_pair.data_refinery.key_name}"
 
-    tag {
-        key = "User"
-        value = "${var.user}"
-        propagate_at_launch = true
+    root_block_device {
+      volume_size = 20
+      volume_type = "gp2"
     }
 
-    tag {
-        key = "Stage"
-        value = "${var.stage}"
-        propagate_at_launch = true
+    tags {
+        Name = "Spot Fleet Launch Specification x1e.16xlarge ${var.user}-${var.stage}"
+        User = "${var.user}"
+        Stage = "${var.stage}"
     }
+  
+  }
+  ##
+  # x1e.32xlarge
+  ##
+  launch_specification {
+
+    # Client Specific
+    instance_type             = "x1.32xlarge"
+    weighted_capacity         = 40 # via https://aws.amazon.com/ec2/instance-types/
+    spot_price                = "${var.spot_price}"
+    ami                       = "${data.aws_ami.ubuntu.id}"
+    iam_instance_profile_arn  = "${aws_iam_instance_profile.data_refinery_instance_profile.arn}"
+    user_data                 = "${data.template_file.nomad_client_script_smusher.rendered}"
+    vpc_security_group_ids    = ["${aws_security_group.data_refinery_worker.id}"]
+    subnet_id                 = "${aws_subnet.data_refinery_1a.id}"
+    availability_zone         = "${var.region}a"
+    key_name = "${aws_key_pair.data_refinery.key_name}"
 
-}
+    root_block_device {
+      volume_size = 20
+      volume_type = "gp2"
+    }
 
-resource "aws_autoscaling_policy" "clients_scale_up" {
-    name = "asg-clients-scale-up-${var.user}-${var.stage}"
-    scaling_adjustment = 1
-    adjustment_type = "ChangeInCapacity"
-    cooldown = 60
-    autoscaling_group_name = "${aws_autoscaling_group.clients.name}"
-    depends_on = ["aws_instance.nomad_server_1"]
-}
+    tags {
+        Name = "Spot Fleet Launch Specification x1e.32xlarge ${var.user}-${var.stage}"
+        User = "${var.user}"
+        Stage = "${var.stage}"
+    }
+  
+  }
+
+  ##
+  # c5d.18xlarge
```

Comment:
C5's are compute optimized, not RAM optimized. I would have thought R5's would be more appropriate here.